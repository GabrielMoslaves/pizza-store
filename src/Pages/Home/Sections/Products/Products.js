import React from "react";
import Card from "../../../../components/Card/Card";
import styles from "./styles.module.scss";
import { useSelector } from "../../../../hooks/useSelector";
import { products } from "../../../../mocks/products";

const Products = () => {
  const { selectedProducts, setSelectedProducts } = useSelector();

  return (
    <div className={styles.container}>
      <h1>Mais vendidas</h1>
      <div className={styles.content}>
        {products.map((item) => (
          <Card
            selectedProducts={selectedProducts}
            setSelectedProducts={setSelectedProducts}
            key={item.id}
            name={item.name}
            description={item.description}
            price={item.price}
            image={item.image}
            product={item}
          />
        ))}
      </div>
    </div>
  );
};

export default Products;
