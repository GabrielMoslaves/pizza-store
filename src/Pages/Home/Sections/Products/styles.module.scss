.container {
  margin-top: 50px;
  margin-bottom: 100px;
  align-items: center;
  display: flex;
  flex-direction: column;
  width: 100%;
  padding: 0 200px;
  font-family: "Sora", sans-serif;
}

.content {
  margin-top: 100px;
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(250px, 1fr));
  justify-content: space-between;
  width: 100%;
  gap: 30px;
}

@media screen and (max-width: 929px) {
  .container {
    padding: 0 20px;
    align-items: center;
  }

  .content {
    display: flex;
    flex-direction: column;
    align-items: center;
    margin-top: 30px;
  }
}
