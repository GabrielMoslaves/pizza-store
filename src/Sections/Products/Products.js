import React from "react";
import Card from "../../components/Card/Card";
import styles from "./styles.module.scss";
import { useSelector } from "../../hooks/useSelector";

const Products = () => {
  const { select, setSelect } = useSelector();

  const products = [
    {
      id: 1,
      name: "La Cuma",
      description: "Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
      price: 23,
      image: "./pizza1.png",
    },
    {
      id: 2,
      name: "La Menta",
      description: "Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
      price: 23,
      image: "./pizza2.png",
    },
    {
      id: 3,
      name: "La Bana",
      description: "Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
      price: 23,
      image: "./pizza3.png",
    },
    {
      id: 4,
      name: "La Toca",
      description: "Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
      price: 23,
      image: "./pizza4.png",
    },
  ];

  console.log({ select });
  return (
    <div className={styles.container}>
      <h1>Mais vendidas</h1>
      <div className={styles.content}>
        {products.map((item) => (
          <Card
            setSelect={setSelect}
            key={item.id}
            name={item.name}
            description={item.description}
            price={item.price}
            image={item.image}
            product={item}
          />
        ))}
      </div>
    </div>
  );
};

export default Products;
