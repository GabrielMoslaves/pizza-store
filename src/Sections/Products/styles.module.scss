.container {
  margin-top: 50px;
  margin-bottom: 100px;
  align-items: center;
  display: flex;
  flex-direction: column;
  width: 100%;
  padding: 0 300px;
  .content {
    margin-top: 100px;
    display: flex;
    justify-content: space-between;
    width: 100%;
    gap: 30px;
  }
}
