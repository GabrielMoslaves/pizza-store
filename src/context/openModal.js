import { createContext, useState } from "react";

export const OpenModalContext = createContext();

export const OpenModalContextProvider = ({ children }) => {
  const [openCartModal, setOpenCartModal] = useState(false);
  const [openNoteModal, setOpenNoteModal] = useState(false);

  return (
    <OpenModalContext.Provider
      value={{ openCartModal, setOpenCartModal, openNoteModal, setOpenNoteModal }}
    >
      {children}
    </OpenModalContext.Provider>
  );
};
