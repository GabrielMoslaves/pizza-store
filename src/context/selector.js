import { createContext, useState } from "react";

export const SelectorContext = createContext();

export const SelectorContextProvider = ({ children }) => {
  const [selectedProducts, setSelectedProducts] = useState([]);

  return (
    <SelectorContext.Provider value={{ selectedProducts, setSelectedProducts }}>
      {children}
    </SelectorContext.Provider>
  );
};
