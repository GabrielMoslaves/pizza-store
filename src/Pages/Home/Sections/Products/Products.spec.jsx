import { render, screen } from "@testing-library/react";
import Products from "./Products";
import { SelectorContextProvider } from "../../../../context/selector";

describe("<Products/>", () => {
  it("should render all products", () => {
    render(
      <SelectorContextProvider>
        <Products />
      </SelectorContextProvider>
    );

    const buttons = screen.getAllByRole("button", { name: /comprar agora/i });
    const images = screen.getAllByRole("img", { name: /la/i });

    expect(buttons.length).toBe(4);
    expect(images.length).toBe(4);
    expect(screen.getByRole("img", { name: /la bana/i })).toHaveAttribute("src", "./pizza3.png");
  });

  it("should match snapshot", () => {
    const { container } = render(
      <SelectorContextProvider>
        <Products />
      </SelectorContextProvider>
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
