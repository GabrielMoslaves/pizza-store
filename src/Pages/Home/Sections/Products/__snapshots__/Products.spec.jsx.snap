// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Products/> should match snapshot 1`] = `
<div
  class="container"
>
  <h1>
    Mais vendidas
  </h1>
  <div
    class="content"
  >
    <div
      class="container"
    >
      <img
        alt="La Cuma"
        src="./pizza1.png"
      />
      <h3>
        La Cuma
      </h3>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit.
      </p>
      <h1>
        R$
        23.99
      </h1>
      <button
        class="container"
      >
        Comprar agora
      </button>
    </div>
    <div
      class="container"
    >
      <img
        alt="La Menta"
        src="./pizza2.png"
      />
      <h3>
        La Menta
      </h3>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit.
      </p>
      <h1>
        R$
        23.99
      </h1>
      <button
        class="container"
      >
        Comprar agora
      </button>
    </div>
    <div
      class="container"
    >
      <img
        alt="La Bana"
        src="./pizza3.png"
      />
      <h3>
        La Bana
      </h3>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit.
      </p>
      <h1>
        R$
        23.99
      </h1>
      <button
        class="container"
      >
        Comprar agora
      </button>
    </div>
    <div
      class="container"
    >
      <img
        alt="La Toca"
        src="./pizza4.png"
      />
      <h3>
        La Toca
      </h3>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit.
      </p>
      <h1>
        R$
        23.99
      </h1>
      <button
        class="container"
      >
        Comprar agora
      </button>
    </div>
  </div>
</div>
`;
